### ISSUE TITLE
Are the <version>-up.js files required?

### ISSUE THREAD

**Issue by hypexr** [PDMLab/massive-migrate#10]
Hello,

I was excited to find this nice little tool, but I'm stuck on having to create the version-up.js files.  I have migration files in s3 and those are just plain SQL like in your examples.  I'd be fine using the dir structure that you outline.  The version-up.js files are a problem for me because I don't want to create them for each new migration.

I want to pull the migrations into a directory and call migrations.runUpMigration. It would apply any new migrations that haven't yet been applied without the version-up.js files. 

Is this reasonable?  Would you be interested in it working this way and interested in doing it or interested in a pull request?

**Comment by AlexZeitler** [comment]
Hello @hypexr, glad you like massive-migrate.

In general we're open for PR, so we should try to find a way to solve your needs.

The reason why the up-files exist is that if you have depedencies inside a particular migration (e.g. first create the table, then fill it with data), you'll be able to ensure the right order of commands being executed.

One solution could be to allow both options. If you provide an `<version>up.js` file, it gets executed, otherwise if you just put `.SQL` files inside the folder, they're just run as is by alpabethical order.

**Comment by hypexr** [comment]
That makes sense.  The alphabetical solution sounds like it would work.  What if instead of a directory structure it's something like:

|-- migrations
|   |-- 1-up-createcustomertable.sql
|   |-- 2-up-createsalutationtable.sql
|   |-- 2-down-createsalutationtable.sql
|-- app.js

The # could be parsed out and used for the run order.

**Comment by AlexZeitler** [comment]
I think we should have a step in between:
Create a folder below `migrations` which can have multiple `.sql` files for a particular migration.

So we have support for:
- One `<version>.SQL` file per migration
- Muitple `.SQL` files per migration in a `<version>` folder executed in alphabetical order
- Have migrations orchestrated by a `<version>-up.js` file (equals current impl.)

Makes sense?

**Comment by hypexr** [comment]
Makes sense.  "One <version>.SQL file per migration" meets my needs so I'll look into adding support for that.

**Comment by AlexZeitler** [comment]
OK, looking forward to your PR.
